lib/serialization/README.md

```diff
@@ -0,0 +1,237 @@
+# Abstract serialization services
+
+The serialization services are centered around the `auto_serializable` annotation,
+the `Serializable` interface and the implementations of `Serializer` and `Deserializer`.
+
+## The `auto_serializable` annotation
+
+A class annotated with `auto_serializable` identifies it as a subclass of Serializable and
+triggers the generation of customized serialization and deserialization services.
+
+~~~
+import serialization
+
+# Simple serializable class identifying a human
+class Person
+	auto_serializable
+
+	# First and last name
+	var name: String
+
+	# Year of birth (`null` if unknown)
+	var birth: nullable Int
+
+	redef fun ==(o) do return o isa SELF and name == o.name and birth == o.birth
+	redef fun hash do return name.hash
```

Comment:
Seeing this two lines here and bellow, I ask myself: can we imagine an `auto_equal` class annotation that care of redifining the `==` and `hash` methods with the same semantic toward attributes than `auto_serialize`. There is also something to thing about cloning, `auto_coloning`; Moreover, each `auto_serialize` class can be easly made `auto_clonable` (the reverse is not true). We can also image `auto_to_s`.

If fact such annotations remind me the `deriving` mechanism of Haskell: http://en.wikibooks.org/wiki/Haskell/Classes_and_types#Deriving